src/core/DY.Core.Trace.Base.fst

```diff
@@ -224,21 +312,33 @@ val entry_exists:
 let entry_exists #label_t tr e =
   exists i. entry_at tr i e
 
+
+/// Is a given entry the last entry on the trace?
+
+val trace_entry_just_occurred: #label_t:Type -> trace_ label_t -> trace_entry_ label_t -> prop
+let trace_entry_just_occurred tr en =
+  is_not_empty tr /\
+  entry_at tr (trace_length tr - 1) en
+  // the more intuitive and cleaner definition:
+  // last tr == en
```

Comment:
nit: this raise the question: "then why not using this better definition?"
I guess the answer is that everything is defined with `entry_at` and proofs will fail if we did so.
A suggestion that seems a bit less suspicious:
```suggestion
  // this is equivalent to:
  // last tr == en
```

Reply:
This was more meant as a comment for discussion:
should we use `last` or stick to `entry_at`?

I prefer `last` but wanted to hear what others think.
If we go for `last`, we probably have to add some lemma relating `last` to `get_entry`

Reply:
Both are equally fine for me, but in any case we certainly need a lemma relating `last` to `get_entry` with an SMT pattern on `last tr`.